homeassistant/components/bayesian/binary_sensor.py

```diff
@@ -185,6 +187,7 @@ def async_threshold_sensor_state_listener(event):
             When a state changes, we must update our list of current observations,
             then calculate the new probability.
             """
+
             new_state = event.data.get("new_state")
 
             if new_state is None or new_state.state == STATE_UNKNOWN:
```

Comment:
This line is causing errors in testing because when a state is removed (equivalent to an entity becoming unavailable) it preserves the previous state which I don't think is desirable and may introduce bugs.
Said another way, if we cannot be sure of the sensor state then we shouldn't be updating our priors.

Yes, this may mean at boot (whilst waiting for sensors to become available) that the the Bayesian sensor may report a different state to that at shutdown, but I think that is OK. (Or at least better than the alternative.

Reply:
Fixed in https://github.com/home-assistant/core/pull/67631/commits/1240df8497d1f96a44fdf04e5ee8d8dda3b15eea